src/newViews/UserGroup/index.tsx

```diff
@@ -0,0 +1,279 @@
+        };
+
+        return ([
+            createStringColumn<Usergroup, number>(
+                'group',
+                _ts('usergroup', 'groupLabel'),
+                item => item.title,
+            ),
+            createNumberColumn<Usergroup, number>(
+                'members',
+                _ts('usergroup', 'membersLabel'),
+                item => item.memberships.length,
+                // to be fetched directly from API as membershipCount when done
+            ),
+            createDateColumn<Usergroup, number>(
+                'createdAt',
+                _ts('usergroup', 'createdOnLabel'),
+                item => item.createdAt,
+            ),
+            actionColumn,
+        ]);
+    }, [handleEditUsergroupClick, usergroupDeleteTrigger]);
+
+    const membersColumns = useMemo(() => {
+        const actionColumn: TableColumn<
+            Membership,
+            number,
+            ActionCellProps<number>,
+            TableHeaderCellProps
+        > = {
+            id: 'action',
+            title: _ts('usergroup', 'actionLabel'),
+            headerCellRenderer: TableHeaderCell,
+            headerCellRendererParams: {
+                sortable: false,
+            },
+            cellRenderer: ActionCell,
+            cellRendererParams: passedUserId => ({
+                itemKey: passedUserId,
+                onEditClick: handleEditMemberClick,
+                onDeleteClick: memberDeleteTrigger,
+                editButtonTitle: _ts('usergroup', 'editMemberLabel'),
+                deleteButtonTitle: _ts('usergroup', 'deleteMemberLabel'),
+                deleteConfirmationMessage: _ts('usergroup', 'deleteMemberConfirmMessage'),
+            }),
+        };
+
+        return ([
+            createStringColumn<Membership, number>(
+                'name',
+                _ts('usergroup', 'nameLabel'),
+                item => item.memberName,
+            ),
+            createStringColumn<Membership, number>(
+                'email',
+                _ts('usergroup', 'emailLabel'),
+                item => item.memberEmail,
+            ),
+            createDateColumn<Membership, number>(
+                'joinedAt',
+                _ts('usergroup', 'addedOnLabel'),
+                item => item.joinedAt,
+            ),
+            createStringColumn<Membership, number>(
+                'role',
+                _ts('usergroup', 'roleLabel'),
+                item => item.role,
+            ),
+            actionColumn,
+        ]);
+    }, [handleEditMemberClick, memberDeleteTrigger]);
+
+    const [rowModifier] = useRowExpansion<Usergroup, number>(
+        ({ datum }) => (
+            <ContainerCard>
+                <Table
+                    className={styles.expandedTable}
+                    columns={membersColumns}
+                    keySelector={membershipKeySelector}
+                    data={datum.memberships}
+                />
+            </ContainerCard>
+        ),
```

Comment:
We may need to define this component separately

Reply:
@frozenhelium @AdityaKhatri what do you think?

Reply:
I think this is okay for now